src/util/newconfig/ConfigFileJson.cpp

```diff
@@ -165,4 +181,97 @@ ConfigFileJson::flattenJson(boost::json::object const& obj, std::string const& p
     }
 }
 
+/*
+{
+    array: [
+        { a: 1},
+        { b: 2}
+    ]
+}
+
+array.[].a
+array.[].b
+
+*/
+
+void
+ConfigFileJson::flattenJsonNonRecursive(boost::json::object const& jsonRootObject)
+{
+    struct Task {
+        boost::json::object const& object;
+        std::string prefix;
+        std::optional<size_t> arrayIndex = std::nullopt;
+    };
+
+    std::queue<Task> tasks;
+    tasks.push(Task{.object = jsonRootObject, .prefix = ""});
+
+    std::unordered_map<std::string, size_t> arraysSizes;
+
+    while (tasks.size() > 0) {
```

Comment:
nit: Queue also provides `.empty()` which probably reads more like english (e.g. `not tasks.empty()`)